diff --git a/gemm.cu b/gemm.cu
index eb0cce7..7d50a1b 100644
--- a/gemm.cu
+++ b/gemm.cu
@@ -59,10 +59,16 @@ void CPU_fill_rand(float *A, int nr_rows_A, int nr_cols_A) {

 int main(int argc, char ** argv){

+  if (argc != 3) {
+    cout << "usage: " << argv[0] << " <size> <repeats>" << endl;
+    return -1;
+  }
+  int size = (int) strtol(argv[1], (char **)NULL, 10);
+  int repeats = (int) strtol(argv[2], (char **)NULL, 10);

-  int min_m_k_n = 2;
-  int max_m_k_n = 4096*4;
-  int repeats = 2;
+  int min_m_k_n = size;
+  int max_m_k_n = size;
+  // int repeats = 20;
   int verbose = 1;

 #ifndef FP16MM
@@ -140,6 +146,7 @@ int main(int argc, char ** argv){
   for(int size = min_m_k_n; size <= max_m_k_n; size=size*2){
     double sum = 0.0;
     for(int rep = 0; rep < repeats; rep++){
+      if (verbose) cout << "repeat: " << rep << endl;
       cudaEventRecord(start, 0);
 	  m=n=k=size;
 	  lda = m;
